web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,444 @@
+  const changeAction = (_, action) => onChange({ device: device.name, action });
+
+  const value = (device.isDrive && action === "resize") ? "keep" : action;
+
+  return (
+    <FormSelect
+      value={value}
+      isDisabled={isDisabled}
+      onChange={changeAction}
+      aria-label={sprintf(_("Space action selector for %s"), device.name)}
+    >
+      <FormSelectOption value="force_delete" label={_("Delete")} />
+      <If
+        condition={!device.isDrive}
+        then={<FormSelectOption value="resize" label={_("Allow resize")} />}
+      />
+      <FormSelectOption value="keep" label={_("Do not modify")} />
+    </FormSelect>
+  );
+};
+
+/**
+ * Row for configuring the space action of a device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ * @param {ProposalSettings} props.settings
+ * @param {number} props.rowIndex - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {number} [props.level=1] - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {number} [props.setSize=0] - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {number} [props.posInSet=0] - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {boolean} [props.isExpanded=false] - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {boolean} [props.isHidden=false] - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {function} [props.onCollapse] - @see {@link https://www.patternfly.org/components/table/#tree-table}
+ * @param {(action: SpaceAction) => void} [props.onChange]
+ */
+const DeviceRow = ({
+  device,
+  settings,
+  rowIndex,
+  level = 1,
+  setSize = 0,
+  posInSet = 1,
+  isExpanded = false,
+  isHidden = false,
+  onCollapse = noop,
+  onChange = noop
+}) => {
+  const treeRow = {
+    onCollapse,
+    rowIndex,
+    props: {
+      isExpanded,
+      isDetailsExpanded: true,
+      isHidden,
+      'aria-level': level,
+      'aria-posinset': posInSet,
+      'aria-setsize': setSize
+    }
+  };
+
+  const spaceAction = settings.spaceActions.find(a => a.device === device.name);
+  const isDisabled = settings.spacePolicy !== "custom";
+  const showAction = !device.partitionTable;
+
+  return (
+    <TreeRowWrapper row={{ props: treeRow.props }}>
+      <Td dataLabel={columnNames.device} treeRow={treeRow}>
+        <DeviceDescriptionColumn device={device} />
+      </Td>
+      <Td dataLabel={columnNames.content}><DeviceContentColumn device={device} /></Td>
+      <Td dataLabel={columnNames.size}><DeviceSizeColumn device={device} /></Td>
+      <Td dataLabel={columnNames.details}><DeviceDetailsColumn device={device} /></Td>
+      <Td dataLabel={columnNames.action}>
+        <If
+          condition={showAction}
+          then={
+            <DeviceActionColumn
+              device={device}
+              action={spaceAction?.action || "keep"}
```

Comment:
should we decide default action at this UI level in table? Maybe we should be strict and return action always from backend code?

Reply:
Well, if the backend does not specify an action then there is no action (it will do nothing with that device). Using "keep" for representing "no action" is something in the side of the UI code.